DatadogSessionReplay/Sources/Feature/RequestBuilder/RequestBuilder.swift

```diff
@@ -97,4 +77,8 @@ internal struct RequestBuilder: FeatureRequestBuilder {
         // Data is already compressed, so request building request w/o compression:
         return builder.uploadRequest(with: multipart.data, compress: false)
     }
+
+    private func url(with context: DatadogContext) -> URL {
+        customUploadURL ?? context.site.endpoint.appendingPathComponent("api/v2/replay")
```

Comment:
a part of me preferred full paths, as in this one it's easy to make a typo

Reply:
Hardcoding the full path lead us to the problem we fix in this PR 🙂. What we do now is: trust the base URL provided by core SDK and specialise it with SR endpoint. Full URLs are asserted in tests, which gives us clarity and discoverability of expected values.

Note that `context.site.endpoint` depends on the Datadog site used.